### ISSUE TITLE
HTTPTargetConnection `getProperties()` throws an error if no `Properties` element is present

### ISSUE THREAD

**Issue by valentine195** [apigee/apigeelint#567]
Apigee Edge [docs ](https://docs.apigee.com/api-platform/reference/api-proxy-configuration-reference#targetendpoint-targetendpointconfiguration) say it is valid to not have a `Properties` element on these, but Apigeelint assumes one will be present.

**Comment by DinoChiesa** [comment]
Hm, @valentine195 Can you give me an example of the failure? I am running it here on a proxy that includes a TargetEndpoint configured with HTTPTargetConnection with no Properties element, and a ServiceCallout with HTTPTargetConnection with no Properties element, and neither case is throwing an error. 

Show me how to reproduce this?

**Comment by valentine195** [comment]
Hey @DinoChiesa, sure.

The following TargetEndpoint and rule cause the error for us:

```xml
<TargetEndpoint name="A">
  <HTTPTargetConnection>
    <URL>http://mocktarget.apigee.net</URL>
    <CommonName>COMMON_NAME_HERE</CommonName>
  </HTTPTargetConnection>
</TargetEndpoint>
```

```ts
export const onTargetEndpoint: EndpointHandler = function (endpoint) {
  const httpTargetConnection = endpoint.getHTTPTargetConnection();
  if (httpTargetConnection) {
    let properties = httpTargetConnection.getProperties();
    if (properties && (properties['request.streaming.enabled'] == 'true' || properties['response.streaming.enabled'] == 'true')) {
      endpoint.addMessage({
        plugin,
        ...plugin,
      });
    }
  }
};
```

Results in: `TypeError: Cannot read properties of undefined (reading 'childNodes')`

It looks like [this line](https://github.com/apigee/apigeelint/blob/d6d100d648dea2996a656a21dc9ea4d49c802d0d/lib/package/HTTPTargetConnection.js#L49-L50) assumes the XPath query will return at least one node, but if there is no `<Properties>` element, it returns an empty list.
